### DISCUSSION TITLE
Ad skipping

### DISCUSSION THREAD

**Discussion by atreyuhere** [dut-iptv/dut-iptv#98] [General]
I only just noticed I can't skip ads in the official Ziggo Go app. Never had any issue in dut-iptv addon.  I think it's bs for a paid service to prevent skip forward , don't know why this is possible in the addon, but i think it's really nice👍
I was thinking, on the flip side of the coin, it might give a provider an added reason not to.....like this addon to be used?

What do you think 🤔

**Comment by dut-iptv** [comment]
Ads (in the cheapest Videoland subscription for example) and several blockades (such as preventing fast forward/rewind/skipping) are implemented in the players that the different providers use in their official apps and web players. 

These ads and limitations are not part of the streams that are used by both the official apps and my addons. It is not possible for my to replicate this functionality. There are ways for the providers to integrate ads in the streams themselves, but no one has chosen to do so at this time. 

So it's an added bonus for users of my addons, but that's just the way these streams are supplied at this time. It's not something I actively created/enabled or promote or plan to promote as part of the README at this time. 

I still believe these addons are pretty low-key and not to many people know about them, which I still kind of like at this time because when the masses and less technically inclined discover these addons I will probably receive a wave of support questions in the issue section. It also keeps these addons of the radar of the providers, although I believe these addons are perfectly legal as outlined in #3. They would still have enough options to prevent the proper functioning of the addons though.

**Reply by kersvanbinsbergen** [reply]
Haha your the best mate.

My thinking was: ziggo can only block the skipping of ads in their app so the stream must be static.   That's why i went looking for a kodi plugin.  

Thank God i found you!

I never mention this plug in on the ziggo forum!  Let's not wake up sleeping dogs :-)

**Comment by atreyuhere** [comment]
I seem to be having some issues with my git account. This session still seems to be ok , though ;)
Anyways, thanks for your reply. I would agree, with the current exposure,  i don't expect too many heads to go up. I sincerely hope it stays that way, as i much prefer using your add-on / Kodi to what Ziggo likes me to use, and i know their stance on third party solutions.

Thanks for explaining how things are enforced within the app. Does the same go, functionality-wise, for the add markers which are present in the Ziggo Go app? Ziggo adds these markers in their streams to mark 'blocks'.
I am only asking, because I remember sometimes seeing these (really convenient) in your add-on as well. Can't seem to now, though..

**Reply by dut-iptv** [reply]
Saw them once or twice too. These chapters are also part of the stream info, so again nothing I am actively doing. It's just part of the stream and it is available or it's not and it isn't available.
